src/core/server/http/router/request.ts

```diff
@@ -182,6 +189,12 @@ export class KibanaRequest<
       // missing in fakeRequests, so we cast to false
       isAuthenticated: Boolean(request.auth?.isAuthenticated),
     };
+
+    this.requestId =
+      this.options.requestOpaqueId.allowFromAnyIp ||
+      this.options.requestOpaqueId.ipAllowlist.includes(this.socket.remoteAddress!)
+        ? request.headers['x-opaque-id'] ?? uuid.v4()
```

Comment:
KibanaRequest is stateless, so we recreate it in route handlers and interceptors.
https://github.com/restrry/kibana/blob/18fe8bbaa622ad7c693b3c08fc6c7b11e78ca2ca/src/core/server/http/router/router.ts#L262
https://github.com/restrry/kibana/blob/18fe8bbaa622ad7c693b3c08fc6c7b11e78ca2ca/src/core/server/http/lifecycle/on_pre_response.ts#L108
etc.
With this approach, every new invocation will generate a new id.

we can:
- create only one KibanaRequest instance across all lifecycles as Hapi does.
- create id once on entry point with `onRequest` interceptor. we could attach it to `Hapi.Request` or store it in a WeakMap.
- reuse `id` generated by Hapi. available under `request.info.id`

Reply:
Fixed, decided to go with the 2nd option by leveraging the writeable `request.app` property.